src/main/telemetry/smartport.c

```diff
@@ -762,8 +762,9 @@ void processSmartPortTelemetry(smartPortPayload_t *payload, volatile bool *clear
             case FSSP_DATAID_T2         :
 #ifdef USE_GPS
                 if (sensors(SENSOR_GPS)) {
-                    // provide GPS lock status
-                    smartPortSendPackage(id, (STATE(GPS_FIX) ? 1000 : 0) + (STATE(GPS_FIX_HOME) ? 2000 : 0) + gpsSol.numSat);
+                    // HDOP 0-9 (0 = 5.5 = lowest, 9 = 1 = highest)
+                    uint8_t hdop = (9 - constrain((gpsSol.hdop - 51) / 50, 0, 9)) * 100;
```

Comment:
This can be expressed in an easier to understand form with `scaleRange()`.

Reply:
Updated using scaleRange().